minvws/nl-kat-coordination
# Feature/boefje normalizer config models

### Changes

Forgive me father, for I have sinned: I did refactoring in my feature branch. Although I think refactoring is often a crucial part of a feature, in this case these were a bit unrelated. Nevertheless, note that I moved the test-directory from the katalogus dir to the central boefjes directory for consistency with the other services (and the fact that we don't have that many test files), and the same holds for the dependencies directory which contains dependencies not strictly related to the KATalogus API as they are used by the boefjes workers as well. Just don't let the amount of files changed distract you: most were just moved.

This feature adds the config models in this overview (as part of [this epic](https://github.com/minvws/nl-kat-coordination/issues/2881)):

```mermaid
classDiagram

class organisation {
  pk
  id
  name
}

class boefje_config {
    id
    settings
    enabled
    boefje_id
    organisation_pk
}

boefje_config --> organisation
boefje_config --> boefje

class normalizer_config {
    id
    enabled
    normalizer_id
    organisation_pk
}

normalizer_config --> organisation
normalizer_config --> normalizer

class boefje {
    id
    plugin_id
    scan_level
    consumes
    produces
    version
    created
    description
    name
    environment_keys
    oci_image
    oci_arguments

    organisations
}

class normalizer {
    id
    plugin_id
    scan_level
    consumes
    produces
    version
    created
    description
    name
    environment_keys

    organisations
}
```

These changes should be backward compatible, as has been tested with migration tests. The API is left intact as well

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/3014

### QA notes

Please verify that the KATalogus works properly, in terms of enabling/disabling and changing/updating/copying settings. Also please check that upgrading to this branch does not disable previously disabled plugins or delete your settings. That shouldn't happen.

---

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
